full-stack-intro - Daniel Yount

TheDanYount opened the pull request.

TheDanYount committed:
Practiced fetching in a full stack env

RobertGardner commented on full-stack-intro/client/src/components/Todos.tsx:
```diff
@@ -0,0 +1,155 @@
+      if (!todoAddResult.ok) throw new Error('add todo failed');
+      const formattedResult = (await todoAddResult.json()) as Todo;
+      const todosCopy = todos.slice();
+      todosCopy.push(formattedResult);
```
This code is good. Personally, I prefer this:
`setTodos([...todos, formattedResult]);`

RobertGardner commented on full-stack-intro/client/src/components/Todos.tsx:
```diff
@@ -0,0 +1,155 @@
+      if (!todoUpdateResult.ok) throw new Error('update todo failed');
+      const newTodos = todos.map((e) => {
+        return e.todoId === todo.todoId ? todo : e;
+      });
```
This code is good. Personally, I prefer to use the shortcut arrow function syntax:
`const newTodos = todos.map((e) => e.todoId === todo.todoId ? todo : e);`

RobertGardner reviewed (approved):
Good! Some possible ways to make the code cleaner -- at least in my opinion! ;-)

TheDanYount committed:
Polish

Used spread and adde a new variable, rather than slice and a push to slice. also used the shortcut for syntax on a 1-line arrow function.
